postgres-protocol/src/message/backend.rs

```diff
@@ -531,11 +532,12 @@ pub struct DataRowBody {
 
 impl DataRowBody {
     #[inline]
-    pub fn ranges(&self) -> DataRowRanges<'_> {
+    pub fn ranges(&self) -> DataRowRanges<'_, DataRow> {
```

Comment:
This module is specifically for parsing the backend protocol messages. I'd rather have composite support added in the types module.